aws/resource_aws_lambda_layer_test.go

```diff
@@ -0,0 +1,215 @@
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSLambdaLayer_basic(t *testing.T) {
+	resourceName := "aws_lambda_layer.lambda_layer_test"
+	layerName := fmt.Sprintf("tf_acc_lambda_layer_basic_%s", acctest.RandString(8))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLambdaLayerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLambdaLayerBasic(layerName),
+				Check:  testAccCheckAwsLambdaLayerExists(resourceName, layerName),
+			},
+
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"filename"},
+			},
+		},
+	})
+}
+
+func TestAccAWSLambdaLayer_s3(t *testing.T) {
+	resourceName := "aws_lambda_layer.lambda_layer_test"
+	rString := acctest.RandString(8)
+	layerName := fmt.Sprintf("tf_acc_lambda_layer_s3_%s", rString)
+	bucketName := fmt.Sprintf("tf-acc-bucket-lambda-layer-s3-%s", rString)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLambdaLayerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLambdaLayerS3(bucketName, layerName),
+				Check:  testAccCheckAwsLambdaLayerExists(resourceName, layerName),
+			},
+
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"s3_bucket", "s3_key"},
+			},
+		},
+	})
+}
+
+func testAccCheckLambdaLayerDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).lambdaconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_lambda_layer" {
+			continue
+		}
+
+		version, err := strconv.Atoi(rs.Primary.Attributes["version"])
+		if err != nil {
+			return err
+		}
+
+		_, err = conn.GetLayerVersion(&lambda.GetLayerVersionInput{
+			LayerName:     aws.String(rs.Primary.ID),
+			VersionNumber: aws.Int64(int64(version)),
+		})
+		if err == nil {
+			return fmt.Errorf("Lambda Layer still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsLambdaLayerExists(res, layerName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[res]
+		if !ok {
+			return fmt.Errorf("Lambda layer not found: %s", res)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Lambda layer ID not set")
+		}
+
+		if rs.Primary.Attributes["version"] == "" {
+			return fmt.Errorf("Lambda layer version not set")
+		}
+
+		version, err := strconv.Atoi(rs.Primary.Attributes["version"])
+		if err != nil {
+			return err
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).lambdaconn
+		_, err = conn.GetLayerVersion(&lambda.GetLayerVersionInput{
+			LayerName:     aws.String(layerName),
+			VersionNumber: aws.Int64(int64(version)),
+		})
+		if err != nil {
```

Comment:
Our future `gosimple` linting will flag this to be simplified to `return err` here. 👍